src/core/__tests__/ReactCompositeComponent-test.js

```diff
@@ -599,6 +599,128 @@ describe('ReactCompositeComponent', function() {
 
   });
 
+  it('should warn if context values differ on update', function() {
+    var Parent = React.createClass({
+      childContextTypes: {
+        foo: ReactPropTypes.string
+      },
+
+      getChildContext: function() {
+        return {
+          foo: "bar"
+        };
+      },
+
+      render: function() {
+        return <div>{this.props.children}</div>;
+      }
+    });
+    var Component = React.createClass({
+      contextTypes: {
+        foo: ReactPropTypes.string.isRequired
+      },
+
+      render: function() {
+        return <div />;
+      }
+    });
+
+    var div = document.createElement('div');
+
+    var componentWithSameContext = React.withContext({foo: 'bar'}, function() {
+      return <Component />;
+    });
+    React.render(<Parent>{componentWithSameContext}</Parent>, div);
+
+    expect(console.warn.mock.calls.length).toBe(0);
+
+    var componentWithDifferentContext = React.withContext({foo: 'noise'}, function() {
+      return <Component />;
+    });
```

Comment:
You're using withContext to test these but withContext can be easily deprecated and quickly removed from the code path.

We should test the scenario where only getChildContext is used.

E.g. just add something like:

```javascript
var Wrapper = React.createClass({

  getChildContext: function() {
    return { foo: 'noise' };
  },

  render: function() { return <Parent><Component /></Parent>; }

});
```